network/server/tchannelthrift/thrift/rpc.thrift

```diff
@@ -50,7 +50,7 @@ service Node {
 	void write(1: WriteRequest req) throws (1: WriteError err)
 	void writeBatch(1: WriteBatchRequest req) throws (1: WriteBatchErrors err)
 	FetchResult fetch(1: FetchRequest req) throws (1: Error err)
-	FetchRawBatchResult fetchRawBatch(1: FetchRawBatchRequest req) throws (1: Error err)
+	FetchRawBatchResult fetchRawBatch(1: FetchRawBatchRequest req)
```

Comment:
For 1) only if there is something other than an application error would FetchRawBatch return `nil` and an error, in all other cases (i.e. partial application error) it will return a `FetchRawBatchResult` with errors for each requested ID captured by the entry for that ID in `FetchRawBatchResult`.

For 2) yes the reason it will always add an error type even though it doesn't throw is that there could always be a transport/protocol/connection error which will be represented as that error if so.